server/webapp/WEB-INF/rails.new/webpack/models/agents/environments.js

```diff
@@ -14,29 +14,34 @@
  * limitations under the License.
  */
 
-const $                = require('jquery');
-const m                = require('mithril');
-const _                = require('lodash');
-const mrequest         = require('helpers/mrequest');
-const TriStateCheckbox = require('models/agents/tri_state_checkbox');
-const Routes           = require('gen/js-routes');
-const Environments     = {};
-Environments.list    = [];
+
+var $                = require('jquery');
+var m                = require('mithril');
+var Stream           = require('mithril/stream');
+var _                = require('lodash');
+var mrequest         = require('helpers/mrequest');
+var TriStateCheckbox = require('models/agents/tri_state_checkbox');
+var Routes           = require('gen/js-routes');
 
 const getSortedEnvironments = (environments, selectedAgents) => {
   const selectedAgentsEnvironments = _.map(selectedAgents, (agent) => agent.environments());
 
   return _.map(environments.sort(), (environment) => new TriStateCheckbox(environment, selectedAgentsEnvironments));
 };
 
-Environments.init = (selectedAgents) => {
-  $.ajax({
-    method:     'GET',
-    url:        Routes.apiv1AdminInternalEnvironmentsPath(),
-    beforeSend: mrequest.xhrConfig.forVersion('v1')
-  }).then((data) => {
-    Environments.list = getSortedEnvironments(data, selectedAgents);
-  }).always(m.redraw);
+var Environments = function () {
```

Comment:
You probably want to call `Environments.all()` to return a list of things. In this case, might as well be an array of (sorted) strings.

Additionally, this still has an issue in that you're not waiting on the ajax promise to complete,